core/src/com/punchbrain/awake/model/Circuit.java

```diff
@@ -23,28 +29,57 @@ public Circuit(GameObject lamp, GameObject flipSwitch, Light lightObject, AudioM
         this.audioManager = audioManager;
     }
 
-    public boolean isOn(){
-        return isOn;
+
+    public Circuit updatePassiveBehaviour(float delta) {
+        updateState(delta);
+        updateAnimationState();
+        if (state == State.ON) {
+            this.lightObject.setDistance(lightRadius);
+        } else {
+            this.lightObject.setDistance(0);
+        }
+        return this;
     }
 
-    public Light getLightObject(){
-        return lightObject;
+    public Circuit resolvePlayerCollision() {
+        // TODO: add player 'push' conditional
+        this.flipSwitch();
+        return this;
     }
 
-    public void flipSwitch(){
-        isOn = !isOn;
-        if (isOn) {
-            audioManager.spawnSound(Assets.Sounds.SWITCH_ON, flipSwitch.getLeft(), flipSwitch.getTop(), 1f, 1f, 200f);
+    private void updateAnimationState() {
```

Comment:
Currently this is named as animation state because the ENUM is in animation, but since it's our primary state for checking the game object, may be worth renaming to `updateGameObjectState` or something